packages/manager/cypress/e2e/core/cloudpulse/edit-firewall-system-alert.spec.ts

```diff
@@ -0,0 +1,419 @@
+    id: 5,
+    label: 'firewall-nodebalancer-5',
+    status: 'enabled',
+    entities: [
+      {
+        id: 5,
+        label: 'nodebalancer-4',
+        type: 'nodebalancer',
+        url: '/test',
+        parent_entity: null,
+      },
+    ],
+  }),
+];
+
+describe('Integration Tests for Edit Alert', () => {
+  /*
+   * - Confirms navigation from the Alert Definitions List page to the Edit Alert page.
+   * - Confirms alert creation is successful using mock API data.
+   * - Confirms that UI handles API interactions and displays correct data.
+   * - Confirms that UI redirects back to the Alert Definitions List page after saving updates.
+   * - Confirms that a toast notification appears upon successful alert update.
+   * - Confirms that UI redirects to the alert listing page after creating an alert.
+   * - Confirms that after submitting, the data matches with the API response.
+   */
+  beforeEach(() => {
+    mockAppendFeatureFlags(flagsFactory.build());
+    mockGetAccount(mockAccount);
+    mockGetFirewalls(mockFirewalls);
+  });
+
+  it('displays and updates firewall alert details for Linode entities', () => {
+    // Navigate to the Edit Alert page
+    const alertDetails = alertFactory.build({
+      description: 'Test description',
+      entity_ids: ['1', '2', '3'],
+      label: 'Alert-1',
+      service_type: 'firewall',
+      severity: 1,
+      status: 'enabled',
+      type: 'system',
+      rule_criteria: {
+        rules: [firewallMetricRulesFactory.build()],
+      },
+    });
+
+    const { id, label, service_type } = alertDetails;
+    mockGetAllAlertDefinitions([alertDetails]).as('getAlertDefinitionsList');
+    mockGetAlertDefinitions(service_type, id, alertDetails).as(
+      'getAlertDefinitions'
+    );
+    mockUpdateAlertDefinitions(service_type, id, alertDetails).as(
+      'updateDefinitions'
+    );
+    cy.visitWithLogin(`/alerts/definitions/edit/${service_type}/${id}`);
+
+    cy.wait('@getAlertDefinitions');
+
+    // Verify that the heading with text 'entity' is visible
+    ui.heading.findByText('entity').should('be.visible');
+
+    // Verify the initial selection of entities, then select all entities.
+    cy.findByText('2 of 2 entities are selected.')
+      .should('be.visible')
+      .closest('[data-qa-notice]')
+      .within(() => {
+        ui.button.findByTitle('Deselect All').should('be.visible').click();
+
+        ui.button
+          .findByTitle('Select All')
+          .should('be.visible')
+          .should('be.enabled');
+      });
+
+    // Confirm notice text updates to reflect selection.
+    cy.findByText('0 of 2 entities are selected.').should('be.visible');
+
+    // Confirm that all invoice items are listed.
+    cy.get('tr').should('have.length', 3);
+
+    cy.get('[data-qa-alert-table-body="true"]').within(() => {
+      cy.get('[data-qa-alert-cell$="_entity"]').then(($cells) => {
+        expect($cells.toArray().map((el) => el.innerText.trim())).to.deep.equal(
```

Comment:
Nit: Let's use proper naming, so it can be readable

```suggestion
        expect($cells.toArray().map((cell) => cell.innerText.trim())).to.deep.equal(
```

Reply:
fixed